mackerel/check.go

```diff
@@ -44,17 +19,31 @@ func (api *API) ReportCheckMonitors(hostID string, reports []*checks.Report) err
 		if len(runes) > messageLengthLimit {
 			msg = string(runes[0:messageLengthLimit])
 		}
-		payload.Reports[i] = &checkReport{
-			Source:               monitorTargetHost{HostID: hostID},
+		payload.Reports[i] = &mkr.CheckReport{
+			Source:               mkr.NewCheckSourceHost(hostID),
 			Name:                 report.Name,
-			Status:               report.Status,
+			Status:               mkr.CheckStatus(report.Status),
 			Message:              msg,
-			OccurredAt:           Time(report.OccurredAt),
-			NotificationInterval: report.NotificationInterval,
-			MaxCheckAttempts:     report.MaxCheckAttempts,
+			OccurredAt:           report.OccurredAt.Unix(),
+			NotificationInterval: normalize(report.NotificationInterval, NotificationIntervalFallback),
+			MaxCheckAttempts:     normalize(report.MaxCheckAttempts, 0),
```

Comment:
The minimum value of `maxCheckAttempts` is `1`.
https://mackerel.io/api-docs/entry/check-monitoring#post
(`0` will be omitted from JSON output and the default value `1` will be used in such a case, so there are no problems, though)